src/RCDEF.CPP

```diff
@@ -609,27 +622,31 @@ int CDEC main( SI argc, char * argv[] )
 	/* check include directory argument if specified */
 	if (HFILESOUT)              /* if argv[6] not NUL */
 	{
-		getcwd( dbuff, 80);
+		dbuff = filesys::current_path();
 		incdir = argv[6];
-		if (chdir(incdir))
+		filesys::path incdirTemp = filesys::path(incdir);
+		filesys::current_path(incdirTemp, pathEC);
+		if (pathEC)
 		{
 			printf( "\nCannot find include directory '%s'.\n", incdir);
 			Errcount++;
 		}
-		chdir(dbuff);
+		filesys::current_path(dbuff);
```

Comment:
Let's go over these changes on a call. I think we can come up with a cleaner solution.